### PR TITLE
tests: drivers: counter: nrf_rtc: fixed_top: Add nrf54l15pdk

### PR DESCRIPTION
**Pull request by nordic-krch**
Follow-up to #71127. nrf54l15 was missing rtc nodes and test was passing because there were no RTCs in the test.

Added nodes, overlay and modified the test to fail if there is no devices to avoid false positives.

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]

**Review by aescolar — DISMISSED** [review]

File: dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi [hunk 1]
```diff
@@ -72,6 +72,16 @@ timer10: timer@85000 {
 	prescaler = <0>;
 };
 
+rtc10: rtc@86000 {
+	compatible = "nordic,nrf-rtc";
+	reg = <0x86000 0x1000>;
```
Comment (anangl) [inline dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi:77]: Could you also add corresponding entries in `validate_base_addresses.c`?

Comment (nordic-krch) [reply dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi:77]: Added.

File: soc/nordic/validate_base_addresses.c [hunk 2]
```diff
@@ -234,7 +234,8 @@ CHECK_DT_REG(rtc0, NRF_RTC0);
 CHECK_DT_REG(rtc1, NRF_RTC1);
 CHECK_DT_REG(rtc2, NRF_RTC2);
 CHECK_DT_REG(rtc130, NRF_RTC130);
-CHECK_DT_REG(rtc131, NRF_RTC131);
```
Comment (anangl) [inline soc/nordic/validate_base_addresses.c:237]: This should not be removed.

**Comment by nordic-krch** [comment]
Closing for now as we don't want to expose RTC on nrf54l15 (at least for now).

**Pull request closed without merging**